File: codes/csharp/object_values.cs
using System; // Console, Func<>, Action<>, Action
using System.Linq; // Enumerable
using System.Collections.Generic; // Dictionary<>, List<>, KeyValuePair<>, IEnumerable<>

class Program {
    static void Main(string[] Args) {
        dynamic IsNumeric = (Func<dynamic, bool>)((dynamic Anything) => (Anything is sbyte || Anything is byte || Anything is short || Anything is ushort || Anything is int || Anything is uint || Anything is long || Anything is ulong || Anything is float || Anything is double || Anything is decimal));

        string JsonStringify(dynamic Anything, bool Pretty = false, string Indent = "    ") {
            int IndentLevel = 0;
            string JsonStringifyInner(dynamic AnythingInner, string IndentInner) {
                if (AnythingInner == null) return "null";
                if (AnythingInner is string) return "\"" + (string)AnythingInner + "\"";
                if ((IsNumeric(AnythingInner) == true) || AnythingInner is bool) return AnythingInner.ToString().Replace(",", ".");
                if (AnythingInner is List<dynamic>) {
                    IndentLevel += 1;
                    string Result = ((Pretty == true) ? ("[" + Environment.NewLine + string.Concat(Enumerable.Repeat(IndentInner, IndentLevel))) : "[");
                    int ArrayItemIndex = 0;
                    foreach (dynamic ArrayItem in (List<dynamic>)AnythingInner) {
                        Result += JsonStringifyInner(ArrayItem, IndentInner);
                        if ((ArrayItemIndex + 1) != ((List<dynamic>)AnythingInner).Count) Result += ((Pretty == true) ? ("," + Environment.NewLine + string.Concat(Enumerable.Repeat(IndentInner, IndentLevel))) : ", ");
                        ArrayItemIndex += 1;
                    }
                    IndentLevel -= 1;
                    Result += ((Pretty == true) ? (Environment.NewLine + string.Concat(Enumerable.Repeat(IndentInner, IndentLevel)) + "]") : "]");
                    return Result;
                }
                if (AnythingInner is Dictionary<string, dynamic>) {
                    IndentLevel += 1;
                    string Result = ((Pretty == true) ? ("{" + Environment.NewLine + string.Concat(Enumerable.Repeat(IndentInner, IndentLevel))) : "{");
                    int ObjectIterationIndex = 0;
                    foreach (KeyValuePair<string, dynamic> ObjectEntry in (Dictionary<string, dynamic>)AnythingInner) {
                        string ObjectKey = ObjectEntry.Key;
                        dynamic ObjectValue = ObjectEntry.Value;
                        Result += "\"" + ObjectKey + "\": " + JsonStringifyInner(ObjectValue, IndentInner);
                        if ((ObjectIterationIndex + 1) != ((Dictionary<string, dynamic>)AnythingInner).Count) Result += ((Pretty == true) ? ("," + Environment.NewLine + string.Concat(Enumerable.Repeat(IndentInner, IndentLevel))) : ", ");
                        ObjectIterationIndex += 1;
                    }
                    IndentLevel -= 1;
                    Result += ((Pretty == true) ? (Environment.NewLine + string.Concat(Enumerable.Repeat(IndentInner, IndentLevel)) + "}") : "}");
                    return Result;
                }
                return "null";
            };
            return JsonStringifyInner(Anything, Indent);
        }

        List<dynamic> ObjectValuesV1(Dictionary<string, dynamic> AnObject) {
            // JavaScript-like Object.values() function
            List<dynamic> NewArray = new List<dynamic>();
            foreach (KeyValuePair<string, dynamic> ObjectEntry in (Dictionary<string, dynamic>)AnObject) {
                string ObjectKey = ObjectEntry.Key;
                dynamic ObjectValue = ObjectEntry.Value;
                NewArray.Add(ObjectValue);
            }
            return NewArray;
        }

        List<dynamic> ObjectValuesV2(Dictionary<string, dynamic> AnObject) {
            // JavaScript-like Object.values() function
            List<dynamic> NewArray = new List<dynamic>();
            foreach (var ObjectValue in AnObject.Values) {
                NewArray.Add(ObjectValue);
            }
            return NewArray;
        }

        Console.WriteLine("\n// JavaScript-like Object.values() in C# Dictionary");

        dynamic Friend = new Dictionary<string, dynamic>() {
            {"name", "Alisa"},
            {"country", "Finland"},
            {"age", 25}
        };
        Console.WriteLine($"Friend: {JsonStringify(Friend, Pretty: true)}");

        Console.WriteLine($"friend values: {JsonStringify(ObjectValuesV1(Friend))}");
        // friend values: ["Alisa", "Finland", 25]

        Console.WriteLine($"friend values: {JsonStringify(ObjectValuesV2(Friend))}");
        // friend values: ["Alisa", "Finland", 25]
    }
}

File: codes/visual_basic_dotnet/object_values.vb
Imports System.Collections.Generic ' Dictionary, KeyValuePair

Module Program
    Function JsonStringify(ByVal Anything As Object, Optional ByVal Pretty As Boolean = False, Optional ByVal Indent As String = "    ") As String
        Dim IndentLevel As Integer = 0
        Dim JsonStringifyInner As Func(Of Object, String, String) = Function(ByVal AnythingInner As Object, ByVal IndentInner As String)
            If (AnythingInner Is Nothing) Then Return "null"
            If (TypeOf AnythingInner Is String) Then Return """" & AnythingInner & """"
            If (IsNumeric(AnythingInner) = True) Then Return CStr(AnythingInner).Replace(",", ".")
            If (TypeOf AnythingInner Is Boolean) Then Return CStr(AnythingInner)
            If (TypeOf AnythingInner Is List(Of Object)) Then
                If (AnythingInner.Count = 0) Then Return "[]"
                IndentLevel += 1
                Dim Result As String = If((Pretty = True), "[" & Environment.NewLine & String.Concat(Enumerable.Repeat(IndentInner, IndentLevel)), "[")
                Dim ArrayItemIndex As Integer = 0
                For Each ArrayItem As Object In AnythingInner
                    Result &= JsonStringifyInner(ArrayItem, IndentInner)
                    If ((ArrayItemIndex + 1) <> AnythingInner.Count) Then Result &= If((Pretty = True), "," & Environment.NewLine & String.Concat(Enumerable.Repeat(IndentInner, IndentLevel)), ", ")
                    ArrayItemIndex += 1
                Next
                IndentLevel -= 1
                Result &= If((Pretty = True), Environment.NewLine & String.Concat(Enumerable.Repeat(IndentInner, IndentLevel)) & "]", "]")
                Return Result
            End If
            If (TypeOf AnythingInner Is Dictionary(Of String, Object)) Then
                If (AnythingInner.Count = 0) Then Return "{}"
                IndentLevel += 1
                Dim Result As String = If((Pretty = True), "{" & Environment.NewLine & String.Concat(Enumerable.Repeat(IndentInner, IndentLevel)), "{")
                Dim ObjectIterationIndex As Integer = 0
                For Each ObjectEntry As KeyValuePair(Of String, Object) In AnythingInner
                    Dim ObjectKey = ObjectEntry.Key
                    Dim ObjectValue = ObjectEntry.Value
                    Result &= """" & ObjectKey & """: " & JsonStringifyInner(ObjectValue, IndentInner)
                    If ((ObjectIterationIndex + 1) <> AnythingInner.Count) Then Result &= If((Pretty = True), "," & Environment.NewLine & String.Concat(Enumerable.Repeat(IndentInner, IndentLevel)), ", ")
                    ObjectIterationIndex += 1
                Next
                IndentLevel -= 1
                Result &= If((Pretty = True), Environment.NewLine & String.Concat(Enumerable.Repeat(IndentInner, IndentLevel)) & "}", "}")
                Return Result
            End If
            Return "null"
        End Function
        Return JsonStringifyInner(Anything, Indent)
    End Function

    Function ObjectValuesV1(ByVal AnObject As Object) As List(Of Object)
        ' JavaScript-like Object.values() function
        Dim NewArray As New List(Of Object)()
        For Each ObjectEntry As KeyValuePair(Of String, Object) In AnObject
            Dim ObjectKey = ObjectEntry.Key
            Dim ObjectValue = ObjectEntry.Value
            NewArray.Add(ObjectValue)
        Next
        Return NewArray
    End Function

    Function ObjectValuesV2(ByVal AnObject As Object) As List(Of Object)
        ' JavaScript-like Object.values() function
        Dim NewArray As New List(Of Object)()
        For Each ObjectValue As Object In AnObject.Values
            NewArray.Add(ObjectValue)
        Next
        Return NewArray
    End Function

    Sub Main(Args As String())
        Console.WriteLine($"{Environment.NewLine}' JavaScript-like Object.values() in Visual Basic (.NET) Dictionary")

        Dim MyFriend As Object = New Dictionary(Of String, Object) From {
            {"name", "Alisa"},
            {"country", "Finland"},
            {"age", 25}
        }
        Console.WriteLine("MyFriend: " & JsonStringify(MyFriend, Pretty:=True))

        Console.WriteLine($"friend values: {JsonStringify(ObjectValuesV1(MyFriend))}")
        ' friend values: ["Alisa", "Finland", 25]

        Console.WriteLine($"friend values: {JsonStringify(ObjectValuesV2(MyFriend))}")
        ' friend values: ["Alisa", "Finland", 25]
    End Sub
End Module
